### PR TITLE
Improve: make gen_version.sh more portable

### PR DESCRIPTION
**Pull request by gpetiot**
Should fix issue #470
It should not have a negative impact on the release procedure.

### CODE REVIEW

File: tools/gen_version.sh [hunk 1]
```diff
@@ -15,12 +15,12 @@
...
 FILE="$1"
 
-if [[ ! -z "$2" ]]; then
+if test -n "$2"; then
     # second arg passed when called from ocamlformat.opam
     VERSION="$2"
 else
     # second arg omitted when called from src/dune
-    if [[ ! "%%VERSION%%" == "%%"*"%%" ]]; then
+    if test "%%VERSION%%" != "%%"*"%%"; then
```
Comment (jberdine) [inline tools/gen_version.sh]: Did you test this? It does not behave the same as before at least in the case when passed only one arg.

Comment (gpetiot) [reply tools/gen_version.sh]: You are right, didn't find a way to make this more portable though

**Comment by jberdine** [comment]
Looking better. Can you describe the testing done to check that the 4 use cases above still work?

**Comment by gpetiot** [comment]
Tests done:
- old gen_version.sh:
  * `tools/gen_version.sh foo` (no watermark); output: `let version = "5e1622b"`
  * `tools/gen_version.sh v42` (no watermark); output: `let version = "v42"`
  * `tools/gen_version.sf foo` (watermarked with 5e1622b); output: `let version = "5e1622b"`
  * `tools/gen_version.sf foo v42` (watermarked with 5e1622b); output: `let version = "v42"`
- new gen_version.sh:
  * `tools/gen_version.sh foo` (no watermark); output: `let version = "5498451"`
  * `tools/gen_version.sh foo v42` (no watermark); output: `let version "v42"`
  * `tools/gen_version.sh foo` (watermarked with 5498451); output: `let version = "5498451"`
  * `tools/gen_version.sh foo v42` (watermarked with 5498451); output: `let version = "v42"`

**Comment by jberdine** [comment]
Ok, thanks, that looks good. I think what I'll do is add this PR to the bottom of the next release branch and try it while preparing the release, and if all goes well then include it in master.

**Comment by hhugo** [comment]
Can we maybe make the script run with bash in master while waiting for this PR to land ?

**Comment by gpetiot** [comment]
Is this PR still relevant now that #592 is merged?

**Comment by jberdine** [comment]
IIRC there were complaints from people who still don't have bash

**Comment by hhugo** [comment]
Bash should be removed from the dune file
https://github.com/ocaml-ppx/ocamlformat/blob/master/src/dune#L15

**Comment by gpetiot** [comment]
Replaced by #664

**Pull request closed without merging**